cpp/src/io/orc/reader_impl.cu

```diff
@@ -610,13 +623,15 @@ reader::impl::impl(std::unique_ptr<datasource> source,
 }
 
 table_with_metadata reader::impl::read(int skip_rows, int num_rows, int stripe,
-                                       int max_stripe_count, cudaStream_t stream) {
+                                       int max_stripe_count,
+                                       const size_type *stripe_indices,
```

Comment:
Why not take a std::vector instead of a count and pointer? The higher-level interface has a vector that can be  directly passed...

Reply:
But the cpp interface doesn't (for somebody using the class api directly), so it would have to create a dummy vector (not that it's very expensive, but why pick the less efficient path ?)

Reply:
I thought the external interface did take a vector. If it doesn't, then why doesn't it?

Reply:
There are 3 cpp APIs that end up in read():
1. read_rows(), takes 2 ints (start row, num rows)
2. read_rowgroup, takes start rowgroup and count
3. read_rowgroups (the newly added version which does take a vector)
In read(), this makes the vector parameter optional, and passing vector::data() as argument allows making this optional (nullptr). Using a constref here would imply having the existing other 2 APIs having to create a local vector vs just passing zero.
The top-level C-like API has a big struct that contains all parameters for all forms of the cpp API, but not at the class level (not something that would be desirable).

Reply:
A default value of an empty vector should eliminate the need for explicitly creating anything when it isn't needed. I doubt you are worried about the performance cost of creating that empty vector. API cleanup doesn't need to happen in this PR, so resolving.